packages/pubsub/lib/transport/transport.ts

```diff
@@ -2,14 +2,16 @@ import type { Event } from '../event.js';
 import type { Result } from '@nangohq/utils';
 
 // TODO: add support for subscriber ack/nack
-export interface Subscription {
-    subject: Event['subject'];
-    callback: (event: Event) => void;
+
+export interface SubscribeProps<TSubject extends Event['subject'] = Event['subject']> {
+    consumerGroup: string;
+    subject: TSubject;
+    callback: (event: Extract<Event, { subject: TSubject }>) => Promise<void> | void;
 }
 
 export interface Transport {
     publish(event: Event): Promise<Result<void>>;
-    subscribe({ consumerGroup, subscriptions }: { consumerGroup: string; subscriptions: Subscription[] }): void;
+    subscribe<TSubject extends Event['subject']>(params: SubscribeProps<TSubject>): void;
```

Comment:
simplifying the arguments as well as making use of the type of the subject to automatically type the event accordingly